fetchcode/api.py

```diff
@@ -27,28 +29,50 @@ class Response:
 - `size`: size of the retrieved content in bytes
 - `url`: fetched URL
     """
-    def __init__(self, location, content_type, size, url):
+    def __init__(self, location, content_type, size, url, scheme):
         self.location = location
         self.content_type = content_type
         self.size = size
         self.url = url
- 
+        self.scheme = scheme
+
 
-def fetch(url):
+def fetch(url, filename=None):
```

Comment:
What is the purpose of `filename`?

Reply:
If user wants to save the file at a customised location, then can specify the location here.

Reply:
Ok, I would change this variable name to something more descriptive, like `location`

Reply:
Agreed, changing the name

Reply:
@majurg , I have changed the name

Reply:
@majurg can you please also tell what is left in this PR :D